docs/mp/integrations/neo4j.adoc

```diff
@@ -0,0 +1,268 @@
+- <<Examples, Examples>>
+- <<Additional Information, Additional Information>>
+- <<References, References>>
+
+== Overview
+
+Neo4j is a graph database management system developed by Neo4j, Inc. It is an ACID-compliant transactional database with native graph storage and processing. Neo4j is available in a GPL3-licensed open-source “community edition”.
+
+include::{rootdir}/includes/dependencies.adoc[]
+
+[source,xml]
+----
+<dependency>
+   <groupId>io.helidon.integrations.neo4j</groupId>
+   <artifactId>helidon-integrations-neo4j</artifactId>
+</dependency>
+----
+
+== Usage
+
+The support for Neo4j is implemented in drivel level. Just add the dependency, add configuration in `microprofile-config.properties` file and Neo4j driver will be configured by Helidon and can be injected using CDI.
+
+First describe Neo4j connection properties:
+
+[source,properties]
+----
+# Neo4j settings
+neo4j.uri=bolt://localhost:7687
+neo4j.authentication.username=neo4j
+neo4j.authentication.password: secret
+neo4j.pool.metricsEnabled: true
+----
+
+Then just inject the driver:
+
+[source,java]
+----
+@Inject
+public MovieRepository(Driver driver) {
+   this.driver = driver;
+}
+----
+
+The driver can be used according to the link:https://neo4j.com/developer/java/[Neo4j documentation].
+
+== Configuration
+
+include::{rootdir}/config/io_helidon_integrations_neo4j.adoc[leveloffset=+1,tag=config]
+
+== Examples
+
+This example implements a simple Neo4j REST service using MicroProfile. For this example a working Neo4j database is required. Neo4j Movie database is used for this example.
+
+Bring up a Neo4j instance via Docker
+
+```bash
+docker run --publish=7474:7474 --publish=7687:7687 -e 'NEO4J_AUTH=neo4j/secret'  neo4j:latest
+```
+
+Goto the Neo4j browser and play the first step of the movies graph: link:http://localhost:7474/browser/?cmd=play&arg=movies[`:play movies`]
+
+Now go to the `pom.xml` and add the following dependencies:
+
+[source,xml]
+----
+ <dependency>
+    <groupId>io.helidon.integrations.neo4j</groupId>
+    <artifactId>helidon-integrations-neo4j</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.neo4j</groupId>
+    <artifactId>helidon-integrations-neo4j-metrics</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.neo4j</groupId>
+    <artifactId>helidon-integrations-neo4j-health</artifactId>
+</dependency>
+----
+
+Next add the connection configuration properties for Neo4j:
+
+[source,properties]
+----
+# Neo4j settings
+neo4j.uri=bolt://localhost:7687
+neo4j.authentication.username=neo4j
+neo4j.authentication.password: secret
+neo4j.pool.metricsEnabled: true
+
+# Enable the optional MicroProfile Metrics REST.request metrics
+metrics.rest-request.enabled=true
+----
+
+this includes both connection information and enables Neo4j metrics propagation.
+
+Finally, we are able to inject and use the `Neo4j` driver. Domain objects (_Movie_, _Actor_, _Person_) code is omitted.
```

Comment:
Not sure what "Domain objects...code is omitted." means here. The example actually refers to all three classes.

Reply:
IMHO, no need to describe obvious data holder objects here.

Reply:
I agree with you and I did not suggest describing those classes. I meant to suggest removing the "Domain objects...code is omitted." sentence. It's clear that the example does not define those classes.

Reply:
removed :)